playbooks/roles/forum/defaults/main.yml

```diff
@@ -19,6 +19,7 @@ FORUM_MONGO_HOSTS:
 FORUM_MONGO_TAGS: !!null
 FORUM_MONGO_PORT: "27017"
 FORUM_MONGO_DATABASE: "cs_comments_service"
+FORUM_MONGO_AUTH_DB: ''
 FORUM_MONGO_USE_SSL: false
 FORUM_MONGO_URL: "mongodb://{{ FORUM_MONGO_USER }}:{{ FORUM_MONGO_PASSWORD }}@{%- for host in FORUM_MONGO_HOSTS -%}{{ host }}:{{ FORUM_MONGO_PORT }}{%- if not loop.last -%},{%- endif -%}{%- endfor -%}/{{ FORUM_MONGO_DATABASE }}{%- if FORUM_MONGO_TAGS -%}?tags={{ FORUM_MONGO_TAGS }}{%- endif -%}"
```

Comment:
@xitij2000, in the instance used for testing this, you have explicitly specified the  `FORUM_MONGO_URL` as

```
FORUM_MONGO_URL: "mongodb://edxforum2:{{ FORUM_MONGO_PASSWORD }}@host1:27017,host2:27017,host3:27017/edxforum2?ssl=true&replicaSet=replicaset-name&authSource=admin&w=majority"
```

This includes additional parameters which are not automatically generated using the existing `FORUM_MONGO_USE_SSL`, newly added `FORUM_MONGO_AUTH_DB` variables. Are these required in the URL as well or is it sufficient to have them set in the `ssl`, `auth_source` variables in `config/mongoid.yml` via the environment variables set using the corresponding `FORUM_MONGO_*` variables?

Reply:
@lgp171188 Sorry that is just a remnant of the old config. The first thing I tried was just supplying these as parameters in the Mongo URI. However, it seems that these are overridden by the other parameters provided by Mongoid (the Ruby library being used for Mongo connections). So I specified them again but didn't remove them from the URI. 

The SSL and AUTH_DB parameters need to go in the environment variables so they can be picked up by the mongoid.yml file, that is the only way it seems to work. 

``w=majority`` isn't required, on edx-platform ``w=1`` and you can't override it so I figure it's for a reason.